File: An2/Sem1/MAP/MAP-Java/MAP-Lab2/src/lab2/Controller/Validator.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package lab2.Controller;

import lab2.Repository.*;
import lab2.Domain.Student;

/**
 *
 * @author mihai
 */
public class Validator {
    private Repository repo;
    
    public Validator(Repository repo){
        /*
         * Constructor
         * uses repository
         */
        this.repo = repo;
    }
    
    public boolean studentExists(int id){
        /*
         * pre cond: int id
         * post cond: true - if student id is found in stack
         *            false - otherwise
         * 
         * checks if a student id was already used
         */
        Student student;
        Stack newStack = new Stack(this.repo.getStackCapacity());
        boolean ok = false;
        while (!ok && this.repo.getStackSize() >-1){
            student = this.repo.getStudent();
            if (student.getId() == id){
                ok = true;
            }
            newStack.push(student);
        }
        while (newStack.getStackSize() > -1){
            this.repo.addStudent(newStack.pop());
        }
        if (ok) return true;
        else return false;
    }
    
    public boolean nameIsEmpty(String name){
        if (name.length() == 0) return true;
        return false;
    }
}

File: An2/Sem1/MAP/MAP-C#/MAP-Lab4/MAP-Lab4/Controller/Validator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using MAP_Lab4.Domain;
using MAP_Lab4.Repository;

namespace MAP_Lab4.Controller
{
    class Validator
    {
        private StudentRepository repo;

        public Validator(StudentRepository repo)
        {
            this.repo = repo;
        }

        public bool studentExists(int id)
        {        
         /*
         * pre cond: int id
         * post cond: true - if student id is found in stack
         *            false - otherwise
         * 
         * checks if a student id was already used
         */

            Student student;
            Stack newStack = new Stack(this.repo.getStackCapacity());
            bool ok = false;
            while (!ok && this.repo.getStackSize() > -1)
            {
                student = this.repo.getStudent();
                if (student.getId() == id)
                    ok = true;
                newStack.push(student);
            }
            while (newStack.getStackSize() > -1)
                this.repo.addStudent(newStack.pop());
            if (ok) return true;
            else return false;
        }

        public bool nameIsEmpty(String name)
        {
            if (name.Length < 1) return true;
            else return false;
        }

    }
}
